maven-plugin/src/main/java/org/openmrs/maven/plugins/Build.java

```diff
@@ -160,13 +187,113 @@ private boolean buildCoreIfWatched(Server server) throws MojoFailureException {
         return false;
     }
 
-    private void buildOwaProject() throws MojoExecutionException {
+    protected void buildOwaProject() throws MojoExecutionException {
+        wizard.showMessage("Building OWA project...");
+
+        boolean isUsingSystemNpmAndNodejs;
+
+        Map<String, String> systemNodeAndNpmVersion = getSystemNodeAndNpmVersion();
+        Map<String, String> batchNodeAndNpmVersion = getBatchNodeAndNpmVersion();
+        if (checkIfVersionsAreDefined(batchNodeAndNpmVersion)) {
+            wizard.showMessage("Using npm and nodejs versions defined in batch mode");
+            if (!batchNodeAndNpmVersion.equals(systemNodeAndNpmVersion)) {
```

Comment:
You can use an existing lib to parse semver versions, see e.g. https://github.com/zafarkhaja/jsemver